guarddog/scanners/github_action_scanner.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import os
+import pathlib
+import typing
+from urllib.parse import urlparse
+
+from guarddog.analyzer.analyzer import Analyzer
+from guarddog.ecosystems import ECOSYSTEM
+from guarddog.scanners.scanner import PackageScanner
+
+log = logging.getLogger("guarddog")
+
+
+class GithubActionScanner(PackageScanner):
+    def __init__(self) -> None:
+        super().__init__(Analyzer(ECOSYSTEM.GITHUB_ACTION))
+
+    def download_and_get_package_info(self, directory: str, package_name: str, version=None) -> typing.Tuple[dict, str]:
+        repo = self._get_repo(package_name)
+        tarball_url = self._get_git_tarball_url(repo, version)
+
+        log.debug(f"Downloading GitHub Action source from {tarball_url}")
+
+        file_extension = pathlib.Path(tarball_url).suffix
+        if file_extension == "":
+            file_extension = ".zip"
+
+        zippath = os.path.join(directory, package_name.replace("/", "-") + file_extension)
+        unzippedpath = zippath.removesuffix(file_extension)
+        self.download_compressed(tarball_url, zippath, unzippedpath)
+
+        return {}, unzippedpath
+
+    def _get_repo(self, url: str) -> str:
+        parsed_url = urlparse(url)
+
+        if parsed_url.hostname and parsed_url.hostname != "github.com":
```

Comment:
This condition can be improved, if the provided url is not correctly parsed (for example if url is `abc`) `urlparse.hostname` will result in `None` and no error will be rised.

```suggestion
        if not parsed_url.hostname or parsed_url.hostname != "github.com":
```

Reply:
I wrote this condition so that it's possible for users to specify both a git URL (https://github.com/actions/checkout.git or https://github.com/actions/checkout) and a repo name (actions/checkout). 
These behaviours are tested in https://github.com/DataDog/guarddog/blob/09fe976f01185b6ad4f37b93c7743bc5c37cb29b/tests/core/test_github_action_scanner.py.